lib/database.js

```diff
@@ -8,54 +8,45 @@ const Schema = require('./schema');
 const SchemaType = require('./schematype');
 const WarehouseError = require('./error');
 const pkg = require('../package.json');
+const { open } = fs.promises;
+const pipeline = Promise.promisify(require('stream').pipeline);
 
-const uncorkAsync = stream => new Promise(resolve => {
-  process.nextTick(() => {
-    stream.uncork();
-    resolve();
-  });
-});
+async function exportAsync(database, path) {
+  const handle = await open(path, 'w');
 
-const exportAsync = Promise.coroutine(function* (database, path) {
-  const stream = fs.createWriteStream(path);
+  try {
+    // Start body & Meta & Start models
+    await handle.write(`{"meta":${JSON.stringify({
+      version: database.options.version,
+      warehouse: pkg.version
+    })},"models":{`);
 
-  // Start body & Meta & Start models
-  stream.write(`{"meta":${JSON.stringify({
-    version: database.options.version,
-    warehouse: pkg.version
-  })},"models":{`);
+    const models = database._models;
+    const keys = Object.keys(models);
+    const { length } = keys;
 
-  const models = database._models;
-  const keys = Object.keys(models);
-  const { length } = keys;
+    // models body
+    for (let i = 0; i < length; i++) {
+      const key = keys[i];
 
-  // support stream backpressure
-  const writeAsync = Promise.promisify(stream.write, { context: stream });
+      if (!models[key]) continue;
 
-  // models body
-  for (let i = 0; i < length; i++) {
-    const key = keys[i];
+      const buffers = [];
 
-    if (!models[key]) continue;
+      if (i) buffers.push(Buffer.from(',', 'ascii'));
 
-    stream.cork();
-    if (i) stream.write(',', 'ascii');
+      buffers.push(Buffer.from(`"${key}":`));
 
-    stream.write(`"${key}":`);
-
-    const modelExportPromise = writeAsync(models[key]._export());
+      buffers.push(Buffer.from(models[key]._export()));
+      await handle.writev(buffers);
```

Comment:
https://github.com/jprichardson/node-fs-extra/blob/3c3865cad8cae7f3046edf434c2ac76abaec7759/lib/fs/index.js#L106-L123

There is a polyfill for `fs.writev` that we could add support for Node.js 10

Reply:
That part is not polyfill, but the code that creates a new wrapper if `fs.writev` actually exists.
Since writev is used for performance purposes, it is not desirable to handle it in alternative code.
After that, there are three choices: `await` in async function in a loop, connect with `Buffer.concat()`, or force call by `WriteStream` internal implementation (implementation before PR).
This time, choose the implementation that requires the least memory copy (it causes a system call instead).